push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/TokenLoader.java

```diff
@@ -167,4 +194,20 @@ public void loadAndQueueTokenBatch(@Observes @Dequeue MessageHolderWithVariants
             }
         }
     }
-}
\ No newline at end of file
+
+    private Set<String> extractTopicsFrom(final Criteria criteria, final String variantID) {
```

Comment:
@secondsun this function is another candidate for the (GCM Topic) Util class. 

I will update the PR...